Story/RainMeadow.StoryHooks.cs

```diff
@@ -49,9 +52,124 @@ private void StoryHooks()
 
             On.Oracle.CreateMarble += Oracle_CreateMarble;
             On.Oracle.SetUpMarbles += Oracle_SetUpMarbles;
+            On.Oracle.SetUpSwarmers += Oracle_SetUpSwarmers;
+            On.OracleSwarmer.BitByPlayer += OracleSwarmer_BitByPlayer;
+            On.SLOracleSwarmer.BitByPlayer += SLOracleSwarmer_BitByPlayer;
+            On.CoralBrain.CoralNeuronSystem.PlaceSwarmers += OnCoralNeuronSystem_PlaceSwarmers;
+            On.SSOracleSwarmer.NewRoom += SSOracleSwarmer_NewRoom;
+            //IL.CoralBrain.CoralNeuronSystem.PlaceSwarmers += ILCoralNeuronSystem_PlaceSwarmers;
 
         }
 
+        private void SSOracleSwarmer_NewRoom(On.SSOracleSwarmer.orig_NewRoom orig, SSOracleSwarmer self, Room newRoom)
+        {
+            if (OnlineManager.lobby == null)
+            {
+                orig(self,newRoom);
+                return;
+            }
+
+            if (!ModManager.MSC)
+            {
+                newRoom.abstractRoom.AddEntity(self.abstractPhysicalObject);
+            }
+        }
+
+        //Only spawn if we own the room
+        private void OnCoralNeuronSystem_PlaceSwarmers(On.CoralBrain.CoralNeuronSystem.orig_PlaceSwarmers orig, CoralBrain.CoralNeuronSystem self)
+        {
+            if (OnlineManager.lobby == null)
+            {
+                orig(self);
+                return;
+            }
+            RoomSession.map.TryGetValue(self.room.abstractRoom, out var room);
+            if (room.isOwner)
+            {
+                orig(self);
+            }
+        }
+
+        //Unused, keeping it around incase SSOracleSwarmer_NewRoom acts up.
```

Comment:
I'd rather not get this merge. Code that turns "you need to own the DLC" into "you need to own the DLC or have our mod" is a no-go.

Reply:
kk removed